File: CPP/PetShop.cpp
//library
#include <iostream>
#include <string>

using namespace std;

//deklarasi class
class PetShop
{
    private:
        int id;         //id 
        string nama;    //nama Produk
        string nkategori; //kategori produk
        string nharga; //harga produk
    
    public:
        //constructor
        PetShop()
        {
            this->id = 0;
            this->nama = "";
            this->nkategori = "";
            this->nharga = "";
        }
        PetShop(int n, string nama, string nkategori, string nharga)
        {
            this->id = n;
            this->nama = nama;
            this->nkategori = nkategori;
            this->nharga = nharga;
        }

        //setter
        void setId(int n)
        {
            this->id = n;
        }
        void setNama(string nama)
        {
            this->nama = nama;
        }
        void setnkategori(string nkategori)
        {
            this->nkategori = nkategori;
        }
        void setnharga(string nharga)
        {
            this->nharga = nharga;
        }

        //getter
        int getId()
        {
            return this->id;
        }
        string getNama()
        {
            return this->nama;
        }
        string getnkategori()
        {
            return this->nkategori;
        }
        string getnharga()
        {
            return this->nharga;
        }

        //methods
        //menambahkan data produk PetShop
        void tambah()
        {
            system("cls");
            cout << "= = = = Menambahkan Data Baru = = = =" << endl;
            cout << "Masukkan ID Anggota PetShop        : ";  
            cin >> this->id;
            cout << "Masukkan Nama produk PetShop       : ";
            cin >> this->nama;
            cout << "Masukkan kategori PetShop          : ";
            cin >> this->nkategori;
            cout << "Masukkan Harga produk              : ";
            cin >> this->nharga;
            cout << "\nData berhasil ditambahkan!" << endl;
            cout << "= = = = = = = = = = = = = = = = = = =\n" << endl;
        }

        //menampilkan data produk PetShop
        void tampil(PetShop data[], int n)
        {
            system("cls");
            cout << "= = =  = List Produk PetShop = = = =" << endl;
            
            if (n == 0) //kalau datanya kosong
            {
                cout << "Kosong \n" << endl;
            }
            else        //kalo ngga tampilin datanya
            {
                for (int i = 0; i < n; i++) 
                {
                    cout << "=================================" << endl;
                    cout << "ID                 : " << data[i].id << endl;
                    cout << "Nama Produk        : " << data[i].nama << endl;
                    cout << "Kategori Produk    : " << data[i].nkategori << endl;
                    cout << "Harga Produk       : " << data[i].nharga << endl;                  
                    cout << "=================================\n" << endl;
                }
            }
        }

        //mengubah data produk PetShop
        void ubah(PetShop data[], int n)
        {
            system("cls");
            tampil(data, n);

            int cari, flag = 0;
            //input user 
            cout << "= = = Update Data produk PetShop = = =" << endl;
            cout << "Masukkan ID produk PetShop yang akan diubah : ";
            cin >> cari;
            //update setiap atribut
            for (int i = 0; i < n; i++)
            {   
                //kalau data id sama dengan yg dicari
                if (cari == data[i].getId())
                {
                    cout << "\n";
                    cout << "Masukkan Nama Produk PetShop        : ";
                    cin >> data[i].nama;
                    cout << "Masukkan kategori produk            : ";
                    cin >> data[i].nkategori;
                    cout << "Masukkan harga produk               : ";
                    cin >> data[i].nharga;
                    cout << "\nData berhasil diubah!" << endl;
                    //flag menjadi 1
                    flag = 1;
                    break;
                }
            }
            //kalau flag != 1 maka
            if (flag != 1)
            {
                cout << "\nID tidak ditemukan!" << endl;
            }
            cout << "= = = = = = = = = = = = = = = = = =\n" << endl;
        }

        //mencari data produk
        void cari (PetShop data[],int n)
        {
            system("cls");
            tampil(data, n);

            int cari, flag = 0;
            //input user
            cout << "= = = = = = = = = = = = = = = = =" << endl;
            cout << "Masukkan ID produk PetShop yang ingin dicari : ";
            cin >> cari;
            //mencari data
            for (int i = 0; i < n; i++)
            {
                //kalau data id sama dengan yg dicari
                if (cari == data[i].getId())
                {
                    cout << "Produk ditemukan dengan ID          : " << data[i].id << endl;
                    cout << "Nama Produk                         : " << data[i].nama << endl;
                    cout << "Kategori Produk                     : " << data[i].nkategori << endl;
                    cout << "Harga Produk                        : " << data[i].nharga << endl;                  
                    cout << "=================================\n" << endl;
                }
            }

        }

        //menghapus data produk PetShop
        void hapus (PetShop data[], int &n)
        {
            system("cls");
            tampil(data, n);

            int cari, flag = 0;
            //input user
            cout << "= = = = = = Hapus data produk PetShop = = = = = =" << endl;
            cout << "Masukkan ID produk PetShop yang ingin dihapus : ";
            cin >> cari;
            //hapus semua atribut
            for (int i = 0; i < n; i++)
            {
                //kalau data id sama dengan yang dicari
                if (cari == data[i].getId())
                {
                    for (int j = i; j < n; j++)
                    {
                        data[j] = data[j+1];
                    }
                    n--;

                    cout << "\nData berhasil dihapus!" << endl;
                    //flag menjadi 1
                    flag = 1;   
                    break;
                }
            }
            //selain 1 maka
            if (flag != 1)
            {
                cout << "\nID tidak ditemukan!" << endl;
            }
            cout << "= = = = = = = = = = = = = = = = = = = = = = = =" << endl;
        }
       
        //destructors
        ~PetShop()
        {

        };
};

File: Java/PetShop.java
import java.util.ArrayList;
import java.util.Scanner;

class PetShop {
    // Atribut
    private int id; // ID produk
    private String nama; // Nama produk
    private String nkategori; // Kategori produk
    private String nharga; // Harga produk

    // Konstruktor default
    public PetShop() {
        this.id = 0;
        this.nama = "";
        this.nkategori = "";
        this.nharga = "";
    }

    // Konstruktor dengan parameter
    public PetShop(int id, String nama, String nkategori, String nharga) {
        this.id = id;
        this.nama = nama;
        this.nkategori = nkategori;
        this.nharga = nharga;
    }

    // Setter methods
    public void setId(int id) {
        this.id = id;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public void setNkategori(String nkategori) {
        this.nkategori = nkategori;
    }

    public void setNharga(String nharga) {
        this.nharga = nharga;
    }

    // Getter methods
    public int getId() {
        return this.id;
    }

    public String getNama() {
        return this.nama;
    }

    public String getNkategori() {
        return this.nkategori;
    }

    public String getNharga() {
        return this.nharga;
    }

    // Method untuk menambahkan data baru
    public void tambah() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("= = = = Menambahkan Data Baru = = = =");
        System.out.print("Masukkan ID Anggota PetShop: ");
        this.id = scanner.nextInt();
        scanner.nextLine(); // Consume newline
        System.out.print("Masukkan Nama produk PetShop: ");
        this.nama = scanner.nextLine();
        System.out.print("Masukkan kategori PetShop: ");
        this.nkategori = scanner.nextLine();
        System.out.print("Masukkan Harga produk: ");
        this.nharga = scanner.nextLine();
        System.out.println("\nData berhasil ditambahkan!");
        System.out.println("= = = = = = = = = = = = = = = = = = =\n");
    }

    // Method untuk menampilkan data
    public void tampil(ArrayList<PetShop> data) {
        System.out.println("= = =  = List Produk PetShop = = = =");
        if (data.isEmpty()) {
            System.out.println("Kosong \n");
        } else {
            for (PetShop item : data) {
                System.out.println("=================================");
                System.out.println("ID                 : " + item.getId());
                System.out.println("Nama Produk        : " + item.getNama());
                System.out.println("Kategori Produk    : " + item.getNkategori());
                System.out.println("Harga Produk       : " + item.getNharga());
                System.out.println("=================================\n");
            }
        }
    }

    // Method untuk mengubah data
    public void ubah(ArrayList<PetShop> data) {
        tampil(data);
        Scanner scanner = new Scanner(System.in);
        System.out.print("Masukkan ID produk PetShop yang akan diubah: ");
        int cari = scanner.nextInt();
        boolean found = false;

        for (PetShop item : data) {
            if (cari == item.getId()) {
                scanner.nextLine(); // Consume newline
                System.out.print("Masukkan Nama Produk PetShop: ");
                item.setNama(scanner.nextLine());
                System.out.print("Masukkan kategori produk: ");
                item.setNkategori(scanner.nextLine());
                System.out.print("Masukkan harga produk: ");
                item.setNharga(scanner.nextLine());
                System.out.println("\nData berhasil diubah!");
                found = true;
                break;
            }
        }

        if (!found) {
            System.out.println("\nID tidak ditemukan!");
        }
        System.out.println("= = = = = = = = = = = = = = = = = =\n");
    }

    // Method untuk mencari data
    public void cari(ArrayList<PetShop> data) {
        tampil(data);
        Scanner scanner = new Scanner(System.in);
        System.out.print("Masukkan ID produk PetShop yang ingin dicari: ");
        int cari = scanner.nextInt();
        boolean found = false;

        for (PetShop item : data) {
            if (cari == item.getId()) {
                System.out.println("Produk ditemukan dengan ID          : " + item.getId());
                System.out.println("Nama Produk                         : " + item.getNama());
                System.out.println("Kategori Produk                     : " + item.getNkategori());
                System.out.println("Harga Produk                        : " + item.getNharga());
                System.out.println("=================================\n");
                found = true;
                break;
            }
        }

        if (!found) {
            System.out.println("ID tidak ditemukan!");
        }
    }

    // Method untuk menghapus data
    public void hapus(ArrayList<PetShop> data) {
        tampil(data);
        Scanner scanner = new Scanner(System.in);
        System.out.print("Masukkan ID produk PetShop yang ingin dihapus: ");
        int cari = scanner.nextInt();
        boolean found = false;

        for (int i = 0; i < data.size(); i++) {
            if (cari == data.get(i).getId()) {
                data.remove(i);
                System.out.println("\nData berhasil dihapus!");
                found = true;
                break;
            }
        }

        if (!found) {
            System.out.println("\nID tidak ditemukan!");
        }
        System.out.println("= = = = = = = = = = = = = = = = = = = = = = = =");
    }
}